nix/default.nix

```diff
@@ -35,38 +35,36 @@ let
 
   # This variable will be passed to Minecraft by PolyMC
   gameLibraryPath = libpath + ":/run/opengl-driver/lib";
+
+  # Use same qt version on polymc and quazip
+  # should be removed when qt6Packages.quazip is available
+  fixedQuazip = quazip.override { inherit qtbase; };
 in
 
-mkDerivation rec {
+stdenv.mkDerivation rec {
   pname = "polymc";
   inherit version;
 
   src = lib.cleanSource self;
 
-  nativeBuildInputs = [ cmake ninja file makeWrapper ];
-  buildInputs = [ qtbase jdk zlib ];
+  nativeBuildInputs = [ cmake ninja jdk file makeWrapper ];
+  buildInputs = [ qtbase fixedQuazip zlib ];
 
   dontWrapQtApps = true;
 
-  postPatch = lib.optionalString (msaClientID != "") ''
-    # add client ID
-    substituteInPlace CMakeLists.txt \
-      --replace '17b47edd-c884-4997-926d-9e7f9a6b4647' '${msaClientID}'
-  '';
-
   postUnpack = ''
-    # Copy submodules inputs
-    rm -rf source/libraries/{libnbtplusplus,quazip}
-    mkdir source/libraries/{libnbtplusplus,quazip}
+    # Copy libnbtplusplus
+    rm -rf source/libraries/libnbtplusplus
+    mkdir source/libraries/libnbtplusplus
     cp -a ${libnbtplusplus}/* source/libraries/libnbtplusplus
-    cp -a ${quazip}/* source/libraries/quazip
-    chmod a+r+w source/libraries/{libnbtplusplus,quazip}/*
+    chmod a+r+w source/libraries/libnbtplusplus/*
   '';
 
   cmakeFlags = [
     "-GNinja"
     "-DLauncher_PORTABLE=OFF"
```

Comment:
The flag is not used anymore, so you can remove it :)
Also, you may be interested in setting the `ENABLE_LTO` one